android/app/src/main/kotlin/coop/polypoly/polypod/ZipTools.kt

```diff
@@ -0,0 +1,55 @@
+package coop.polypoly.polypod.polyNav
+
+import android.content.Context
+import androidx.security.crypto.EncryptedFile
+import androidx.security.crypto.MasterKey
+import java.io.File
+import java.io.InputStream
+import java.util.zip.ZipInputStream
+
+class ZipTools {
+    companion object {
+        open fun unzipAndEncrypt(
+            inputZipStream: InputStream,
+            context: Context,
+            folder: String
```

Comment:
Isn't there a type in Kotlin that ensures that the folder exists and it's a real folder?

Reply:
Sure, there's `Path` and `File`, but here it isn't really a full path to folder. This is only a name of a folder to create.